server/controller/taskController.js

```diff
@@ -1,59 +1,65 @@
 const taskService = require('../service/taskService');
 
 const taskController = {
-    getTasks: (req, res) => {
+    getTasks: (req, res, next) => {
         taskService.getTasks(req.user)
             .then(result => {
                 res.status(200).json({ result });
             })
             .catch(err => {
-                res.status(500).json({ message: 'Something went wrong.' });
+                //res.status(500).json({ message: 'Something went wrong.' });
+                next(err)
             })
     },
-    getTask: (req, res) => {
+    getTask: (req, res, next) => {
         taskService.getTask(req.params.id, req.user)
             .then(result => {
                 res.status(200).json({ result });
             })
             .catch(err => {
-                res.status(500).json({ message: 'Something went wrong.' });
+                next(err);
+                //res.status(500).json({ message: 'Something went wrong.' });
             })
     },
-    addTask: (req, res) => {
+    addTask: (req, res, next) => {
```

Comment:
please remove all next parameters from your controllers.
you need to use res.handleError(error)